### DISCUSSION TITLE
Is it possible to create pagination like in google docs?

### DISCUSSION THREAD

**Discussion by lumenwrites** [ueberdosis/tiptap#2249] [General Questions] [answered]
Hey guys! It would be truly amazing if it were possible to create google-docs-like pagination.

Do you think it's possible to accomplish with TipTap?

**Comment by philippkuehn** [comment, marked as answer]
See https://github.com/ueberdosis/tiptap/issues/1870

**Reply by rezaffm** [reply]
this feature would be indeed phenomenal - I had a look at this plugin, but couldn't get it work: https://www.npmjs.com/package/prosemirror-pagination maybe there is a chance to get in touch with the guy who initially coded it - something like this is far out of my scope, I am actually happy that I understood the basics of Prosemirror at all.

**Reply by hanspagel** [reply]
I’m with Marijn, the creator of ProseMirror and think the same applies to Tiptap:
> There’s some related discussion in this thread 205, but this is definitely not something that ProseMirror supports—it tries to be an editor of semantic content, and paging is very much a presentation issue that, in my opinion, should be separated from the content editing process.

https://discuss.prosemirror.net/t/paging-document/1829/2

**Reply by rezaffm** [reply]
Well, I could also live with that but how could something like this be achieved with vue? Basically you need a pagination component and several instances of tiptap (editors basically?) and the menu bar is then used for the respective instances?

**Reply by hanspagel** [reply]
I’d always optimise for the medium the content is presented on and if someone wants to print stuff, the printer is doing a good job with splitting long content into separate pages. I don't see a reason to present content as sheets of paper on a screen. And if people want to do that, that’s just not what we have to solve, that’s what I think.

**Reply by rezaffm** [reply]
It is less about presentation for me Hans. We use LanguageTool and even with other editors if words count goes over > 2.000 things start to get slower. Using a google doc is not an issue, as they have per page different instances.

**Reply by aehlke** [reply]
If you think about pagination in terms of a slide deck, it's IMO odd to say that the content that belongs to each slide is not a semantic concern, only a presentation issue. In this sense page breaks are semantic.

**Reply by shmish111** [reply]
I don't really understand how pagination is not semantic? We are creating an editor in which users will want to know what their final PDF will look like as they type, it's super important to know what page something is on and to be able to control that. In HTML and CSS we have plenty of support for that but it's when it's static. In the editor we need something that can dynamically add something like an hr node into some text based on the height since the last hr. I suppose there's a chance that this is not possible considering that the height of a node depends on the styling? Perhaps we can use javascript to work out the height and tell this to tiptap constantly? Does this sound like a solution I would be able to write as an extension? Or does anyone have any ideas of the basic mechanics of how an extension could achieve this?

**Reply by aehlke** [reply]
Perhaps this could help https://github.com/pagedjs/pagedjs

**Reply by dan-cooke** [reply]
@aehlke I fully agree with @shmish111 . I really don't see how its not semantic. Its maybe not semantic if your use case is to produce an HTML document. But if your final output is a report or a letter - pages are absolutely crucial and should be considered part of the content.

paged.js is also only about print previews

**Reply by RomikMakavana** [reply]
I am working on tiptap, to build many rich text editor and my custom extensions, past 3 years. and successfully archive all of them.
But with pagination, I loss hope to build it, there is some static reasons.
I have explained here https://medium.com/@romik-mk/tiptap-pagination-problems-solutions-31f1a0b51e08

Also, doing some research It is all way possible implementing, 
Joke : Like cleaning 99% germs (bugs)
I am willing to make it open source, in sometime.

Thank you

**Reply by RomikMakavana** [reply]
Hello everyone
Here is complete pagination
https://romikmakavana.me/tiptap-pagination/

Thank you

**Reply by flpsnd** [reply]
@RomikMakavana you're a legend mate, apprecite your work🙏

**Comment by xiangshu233** [comment]
@shmish111 Hello, have you found out about a rich text editor that supports pagination? I also searched the Internet and found it here. It seems that none of the rich text editors on the market support pagination. I want the word one. Effect

**Comment by adityayaduvanshi** [comment]
Please try my tiptap  pagination npm package. [https://www.npmjs.com/package/tiptap-pagination-breaks](https://www.npmjs.com/package/tiptap-pagination-breaks)

**Reply by xiangshu233** [reply]
It is recommended to add an online demo or video

**Reply by RomikMakavana** [reply]
https://romikmakavana.me/tiptap-pagination/

**Comment by RomikMakavana** [comment]
Here is tiptap pagination with example
https://romikmakavana.me/tiptap-pagination/

Thank you
